packages/mds-cache/index.ts

```diff
@@ -198,10 +198,28 @@ async function hwrite(suffix: string, item: CacheReadDeviceResult | Telemetry |
     await (await getClient()).hdelAsync(key, ...nulls)
   }
 
-  await (await getClient()).hmsetAsync(key, hmap)
+  const client = await getClient()
+
+  if (suffix === 'event') {
+    await client.hmsetAsync(`provider:${item.provider_id}:latest_event`, hmap)
+  }
+  await client.hmsetAsync(key, hmap)
   return updateVehicleList(device_id)
 }
 
+async function getMostRecentEventByProvider(): Promise<{ provider_id: string; max: number }[]> {
+  const client = await getClient()
+  const result = await client.hgetallAsync('provider:*:latest_event')
+  const payload: { provider_id: string; max: number }[] = []
+  for (const key of Object.keys(result)) {
+    const [, provider_id] = key.split(':')
+    const item: CacheReadDeviceResult | Telemetry | VehicleEvent = JSON.parse(result[key])
+    const max = item.recorded || 0
+    payload.push({ provider_id, max })
+  }
```

Comment:
Seems like this would be a good place to use `Object.keys(result).map(key => ...)`